xarray/core/dataarray.py

```diff
@@ -2244,10 +2256,40 @@ def reset_index(
         See Also
         --------
         DataArray.set_index
+        DataArray.set_xindex
+        DataArray.drop_indexes
         """
         ds = self._to_temp_dataset().reset_index(dims_or_levels, drop=drop)
         return self._from_temp_dataset(ds)
 
+    def set_xindex(
+        self,
+        coord_names: Hashable | Sequence[Hashable],
+        index_cls: type[Index],
+        **options,
+    ) -> DataArray:
```

Comment:
```suggestion
    ) -> T_DataArray:
```
If you have a subclass of DataArray I think you should get the same thing back. It's odd that mypy isn't complaining here.

Reply:
Honestly I did the same than `DataArray.reset_index` here after mypy was complaining for `Dataset.set_xindex`, without thinking too much and without looking at the referenced mypy issue more in depth.

That's all very confusing to me :)

Reply:
Might be blocked again by https://github.com/python/mypy/issues/12846
Which is annoying since mypy produces false positives on the user side...